ray-operator/controllers/ray/utils/util.go

```diff
@@ -201,10 +201,13 @@ func CalculateMaxReplicas(cluster *rayiov1alpha1.RayCluster) int32 {
 	return count
 }
 
-// CalculateDesiredReplicas calculate desired worker replicas at the cluster level
+// CalculateAvailableReplicas calculates available worker replicas at the cluster level
 func CalculateAvailableReplicas(pods corev1.PodList) int32 {
 	count := int32(0)
 	for _, pod := range pods.Items {
+		if val, ok := pod.Annotations["ray.io/node-type"]; !ok || val != string(rayiov1alpha1.WorkerNode) {
```

Comment:
I had to use `ray.io/node-type` because importing `common.RayNodeTypeLabelKey` caused import cycle.

```
-: import cycle not allowed: import stack: [
  github.com/ray-project/kuberay/ray-operator
  github.com/ray-project/kuberay/ray-operator/controllers/ray
  github.com/ray-project/kuberay/ray-operator/controllers/ray/batchscheduler
  github.com/ray-project/kuberay/ray-operator/controllers/ray/batchscheduler/volcano
  github.com/ray-project/kuberay/ray-operator/controllers/ray/common
  github.com/ray-project/kuberay/ray-operator/controllers/ray/utils
  github.com/ray-project/kuberay/ray-operator/controllers/ray/common
]
```

Reply:
Cool. Maybe we need to separate `constant.go` into a separate module in the future. cc @DmitriGekhtman @Jeffwan @architkulkarni

Reply:
Yeah, we should track solving that.
It would be better if we could eliminate common and utils by refactoring the package structure -- https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common